### ISSUE TITLE
IP duplication: calico assigns the same IP to tunnel interface and pod

### ISSUE THREAD

**Issue by VitaliiBabii** [projectcalico/calico#6244]
Sometimes(at least one per day) there is cases when calico on the same node assigns the same IP address to both Tunnel Interface and Pod. As a result network connection(routes) from inside the Pod with duplicated IP is broken. After manual pod deletion new pod gets new IP and works as expected.

## Expected Behavior
Calico assigns unique IP to Tunnel and Pod. No IP duplicates.

## Current Behavior
Sometimes calico assigns the same IP to tunnel interface and pod. 

## Steps to Reproduce (for bugs)
It's hard to reproduce. We have self-managed cluster with about 1000 nodes in AWS. Workload is very dynamic, lots of jobs/pods starting/terminating . Cluster scales up and down all the time. 
Probably there could be some race condition in some of calico components when old node get terminated and new one starting with the same aws internal IP(hostname).

## Environment
Calico v3.21.5
Kubernetes 1.21.10
CRI-O v1.21.6
Operating System and version: Rocky Linux release 8.6
Kernel: 4.18.0-372.9.1.el8.x86_64

Calico logs from affected node and calico-kube-controllers attached. Affected node hostname(IP) in the log is ip-10-116-10-4.eu-west-1.compute.internal and duplicated IP 100.111.85.73.

'ip a' output on node  ip-10-116-10-4.eu-west-1.compute.internal:

```
5: tunl0@NONE: <NOARP,UP,LOWER_UP> mtu 1440 qdisc noqueue state UNKNOWN group default qlen 1000
    link/ipip 0.0.0.0 brd 0.0.0.0
    inet 100.111.85.73/32 scope global tunl0
       valid_lft forever preferred_lft forever 
```
One of pods with duplicated IP:

<img width="1517" alt="podIP" src="https://user-images.githubusercontent.com/17087859/174541304-e9b6455d-6d0f-4c66-9520-5a2faca3606b.png">

[calico-kube-controllers-648fff4675-8xxnj.zip](https://github.com/projectcalico/calico/files/8938149/calico-kube-controllers-648fff4675-8xxnj.zip)
[calico-node-nft-9tzj7.zip](https://github.com/projectcalico/calico/files/8938202/calico-node-nft-9tzj7.zip)

**Comment by VitaliiBabii** [comment]
`calicoctl ipam check` output(non important lines omitted):

```
Loading all IPAM pools...
  100.96.0.0/11
Found 1 active IP pools.

Loading all nodes.
Found 797 node tunnel IPs.

Loading all workload endpoints.
Found 15068 workload IPs.
Workloads and nodes are using 15865 IPs.

Found 4245 IPs that are allocated in IPAM but not actually in use.
.....................
Scanning for IPs that are in use by a workload or node but not allocated in IPAM...
  100.111.85.73 in use by Node(ip-10-116-10-4.eu-west-1.compute.internal) and in active IPAM pool but has no IPAM allocation.

Found 0 in-use IPs that are not in active IP pools.

Found 1 in-use IPs that are in active IP pools but have no corresponding IPAM allocation.
```

**Comment by caseydavenport** [comment]
@VitaliiBabii hello, I'm the author of that fix and can confirm it was included in v3.22.2.

It does sound like you might be experiencing the same issue, which was a race condition when adding / deleting nodes in the cluster, although I can't say for sure that it's the same root cause. I think it would make sense to upgrade to v3.22.2 or later and see if that works better.

**Comment by VitaliiBabii** [comment]
@caseydavenport is there a chance that fix https://github.com/projectcalico/calico/pull/5613 will be added to v3.21.х as a patch ?

**Comment by caseydavenport** [comment]
There aren't any plans for that currently to my knowledge. It's a big change to back-port that far, so I think it's unlikely.

**Comment by caseydavenport** [comment]
Closing as the fix is in v3.22.2 and later.
